core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -306,6 +312,98 @@ default Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessi
                 .iterator();
     }
 
+    /**
+     * Gets comments for all relations (tables, views, materialized views) that match the specified prefix.
+     * Redirected table names are included, but the comment for them is not.
+     */
+    default Iterator<TableCommentMetadata> streamRelationComments(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        Map<SchemaTableName, TableCommentMetadata> materializedViews;
+        Map<SchemaTableName, TableCommentMetadata> views;
+        List<TableCommentMetadata> tables;
+
+        if (prefix.getTable().isPresent()) {
+            SchemaTableName tableName = prefix.toOptionalSchemaTableName().orElseThrow();
+            Optional<ConnectorMaterializedViewDefinition> materializedView = getMaterializedView(session, tableName);
+            if (materializedView.isPresent()) {
+                materializedViews = Map.of(tableName, TableCommentMetadata.forTable(tableName, materializedView.get().getComment()));
+                views = Map.of();
+                tables = List.of();
+            }
+            else {
+                materializedViews = Map.of();
+                Optional<ConnectorViewDefinition> view = getView(session, tableName);
+                if (view.isPresent()) {
+                    views = Map.of(tableName, TableCommentMetadata.forTable(tableName, view.get().getComment()));
+                    tables = List.of();
+                }
+                else {
+                    views = Map.of();
+                    Optional<CatalogSchemaTableName> redirected = redirectTable(session, tableName);
+                    if (redirected.isPresent()) {
+                        tables = List.of(TableCommentMetadata.forRedirectedTable(tableName));
+                    }
+                    else {
+                        ConnectorTableHandle tableHandle = getTableHandle(session, tableName, Optional.empty(), Optional.empty());
```

Comment:
filter with `accessControl` before calling `getMaterializedView`, `getView`, getTableHandle` ... or maybe engine should do that before invoking the connector

see also https://github.com/trinodb/trino/pull/18515